src/proto/streams/flow_control.rs

```diff
@@ -61,6 +61,11 @@ impl FlowControl {
         self.window_size.as_size()
     }
 
+    /// Check if send window is available
+    pub fn window_available(&mut self, sz: WindowSize) -> bool {
```

Comment:
The name of this function could be confusing, since there is both `window_size()` and `available()`, so a function `window_available()` sounds like a combination, while at the same time possible a question. I'd suggest leaving this function off, and just doing `if flow.window_size() >= sz` wherever it's needed.